src-ghc/Pact/Bench.hs

```diff
@@ -162,15 +179,49 @@ mkBenchCmd kps (str, t) = do
 pk :: Text
 pk = "0c99d911059580819c6f39ca5c203364a20dbf0a02b0b415f8ce7b48ba3a5bad"
 
+perfEnv :: Text -> PerfTimer -> PactDbEnv p -> PactDbEnv p
+perfEnv n pt (PactDbEnv db mv) = PactDbEnv (perfPactDb n pt db) mv
+
+perfInterpreter :: Text -> PerfTimer -> Interpreter e -> Interpreter e
+perfInterpreter msg pt (Interpreter i) = Interpreter $ \runInput ->
+  perf pt (msg <> ":" <> "interp") $! i runInput
+
+
+mkFilePerf :: FilePath -> IO PerfTimer
+mkFilePerf fp = do
+  h <- openFile fp WriteMode
+  c <- newChan
+  void $ forkIO $ forever $ do
+    m <- getChanContents c
+    (`mapM_` m) $ \s -> do
+      hPutStrLn h s
+      hFlush h
+  return $ PerfTimer $ \msg a -> do
+    s <- liftIO $ time
+    r <- a
+    liftIO $ do
+      e <- time
+      writeChan c $! unpack $ msg <> ": " <> pack (show (e .-. s))
+      --f <- time
+      --writeChan c $! "chan: " <> (show (f .-. e))
+      return r
+
+  where
+    time = getCurrentTime
+
 main :: IO ()
 main = do
+  print =<< getNumCapabilities
+  !fperf <- mkFilePerf "pact-bench-perf" -- foo
+  let !dbpt = if doPerf == Db || doPerf == All then fperf else def
```

Comment:
Not crucial, but the naming of `dbpt` and `ipt` is a little confusing

Reply:
will rename